test/e2e/ginkgo/multiple_cluster_deployment_test.go

```diff
@@ -0,0 +1,305 @@
+		})
+
+		ginkgo.It("handles changes in bindingpolicy ObjectSelector", func() {
+			ginkgo.By("deletes WEC objects when bindingpolicy ObjectSelector stops matching")
+			patch := []byte(`{"spec": {"downsync": [{"objectSelectors": [{"matchLabels": {"app.kubernetes.io/name": "invalid"}}]}]}}`)
+			_, err := ksWds.ControlV1alpha1().BindingPolicies().Patch(
+				ctx, "nginx", types.MergePatchType, patch, metav1.PatchOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 0)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+
+			ginkgo.By("creates WEC objects when bindingpolicy ObjectSelector matches")
+			patch = []byte(`{"spec": {"downsync": [{"objectSelectors": [{"matchLabels": {"app.kubernetes.io/name": "nginx"}}]}]}}`)
+			_, err = ksWds.ControlV1alpha1().BindingPolicies().Patch(
+				ctx, "nginx", types.MergePatchType, patch, metav1.PatchOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 1)
+			util.ValidateNumDeployments(ctx, wec2, ns, 1)
+		})
+
+		ginkgo.It("deletes WEC objects following deletion of the bindingpolicy object in wds", func() {
+			err := ksWds.ControlV1alpha1().BindingPolicies().Delete(ctx, "nginx", metav1.DeleteOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+
+			util.ValidateNumDeployments(ctx, wec1, ns, 0)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+		})
+
+		ginkgo.It("handles multiple bindingpolicies with overlapping matches", func() {
+			ginkgo.By("creates a second bindingpolicy with overlapping matches")
+			util.CreateBindingPolicy(ctx, ksWds, "nginx-2",
+				[]metav1.LabelSelector{
+					{MatchLabels: map[string]string{"location-group": "edge"}},
+				},
+				[]ksapi.DownsyncObjectTest{
+					{ObjectSelectors: []metav1.LabelSelector{
+						{MatchLabels: map[string]string{"app.kubernetes.io/name": "nginx"}},
+					}}})
+			util.ValidateNumDeployments(ctx, wec1, ns, 1)
+			util.ValidateNumDeployments(ctx, wec2, ns, 1)
+
+			ginkgo.By("delete the second bindingpolicy")
+			err := ksWds.ControlV1alpha1().BindingPolicies().Delete(ctx, "nginx-2", metav1.DeleteOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 1)
+			util.ValidateNumDeployments(ctx, wec2, ns, 1)
+		})
+
+		ginkgo.It("deletes WEC objects when wds deployment is deleted", func() {
+			util.DeleteDeployment(ctx, wds, ns, "nginx")
+			util.ValidateNumDeployments(ctx, wec1, ns, 0)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+		})
+
+		ginkgo.It("deletes WEC objects when BindingPolicy is deleted", func() {
+			err := ksWds.ControlV1alpha1().BindingPolicies().Delete(ctx, "nginx", metav1.DeleteOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 0)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+		})
```

Comment:
Isn't the same test above in lines 86--92?

Reply:
Yes. Good catch. Don't know how it snack in there.

Reply:
The duplication has not been removed yet.

Reply:
fixed.